Compiling use_context hook example fails

<!-- Please only include one item in each issue! -->

This is about:
- [ ] A typo 
- [x] Inaccurate/misleading documentation (e.g. technically incorrect advice)
- [ ] Undocumented code
- [ ] Outdated documentation
- [ ] Other

**Problem**
<!-- PLEASE INCLUDE A WAY TO LOCATE WHERE THE ISSUE IS, e.g.
 - A Github link to the text
 - The filename and line number
 - An excerpt from the docs that someone could use text search to locate
-->
<!-- What's wrong? -->
The `use_context` hook example does not work.
[https://yew.rs/docs/en/next/concepts/function-components/pre-defined-hooks](https://yew.rs/docs/en/next/concepts/function-components/pre-defined-hooks)

**Steps To Reproduce**
1. Create a Yew project
2. In `Cargo.toml` add dependencies
```
yew = {git = "https://github.com/yewstack/yew"}
yew-functional = {git = "https://github.com/yewstack/yew"}
yew-functional-macro = {git = "https://github.com/yewstack/yew"}
```
2.  Copy paste example code into app.rs
5. Add the following imports to app.rs
```
use yew::html;
use yew_functional::{function_component, use_state, ContextProvider, use_context};
use std::rc::Rc;
```
6. The compiler gives the following error
```
error[E0277]: the trait bound `yew::virtual_dom::VComp: Transformer<Rc<Theme>, Theme>` is not satisfied
  --> src/app.rs:19:41
   |
19 |         <ContextProvider<Theme> context=ctx>
   |                                         ^^^ the trait `Transformer<Rc<Theme>, Theme>` is not implemented for `yew::virtual_dom::VComp`
   |
   = help: the following implementations were found:
             <yew::virtual_dom::VComp as Transformer<&'a T, Option<T>>>
             <yew::virtual_dom::VComp as Transformer<&'a T, T>>
             <yew::virtual_dom::VComp as Transformer<&'a str, Option<String>>>
             <yew::virtual_dom::VComp as Transformer<&'a str, String>>
           and 3 others
   = note: required by `transform`

```


**Questionaire** _(Optional)_
- [ ] I'd like to write this documentation
- [ ] I'd like to write this documentation but I'm not sure what's needed
- [x] I don't have time to add this right now, but maybe later

--- a/docs/concepts/function-components/pre-defined-hooks.md
+++ b/docs/concepts/function-components/pre-defined-hooks.md
@@ -293,7 +293,8 @@ pub fn app() -> Html {
     });
 
     html! {
-        <ContextProvider<Theme> context=ctx>
+        // `ctx` is type `Rc<Theme>` while we need `Theme` so we deref it
+        <ContextProvider<Theme> context=(*ctx).clone()>
             // Every child here and their children will have access to this context.
             <Toolbar />
         </ContextProvider<Theme>>
@@ -315,7 +316,7 @@ pub fn toolbar() -> Html {
 /// As this component is a child of `ThemeContextProvider` in the component tree, it also has access to the context.
 #[function_component(ThemedButton)]
 pub fn themed_button() -> Html {
-    let theme = use_context::<Rc<Theme>>().expect("no ctx found");
+    let theme = use_context::<Theme>().expect("no ctx found");
 
     html! {
         <button style=format!("background: {}; color: {};", theme.background, theme.foreground)>

